pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,297 @@
+					} else {
+						if err := processProfileChange(getconfigCtx, localProfile); err != nil {
+							log.Errorf("localProfileTimerTask processProfileChange: %s", err)
+						}
+					}
+				}
+				ctx.ps.CheckMaxTimeTopic(wdName, "getLocalProfileConfig", start,
+					warningTime, errorTime)
+			}
+
+		case <-stillRunning.C:
+		}
+		ctx.ps.StillRunning(wdName, warningTime, errorTime)
+	}
+}
+
+func parseLocalProfile(localProfileBytes []byte) (*profile.LocalProfile, error) {
+	var localProfile = &profile.LocalProfile{}
+	err := proto.Unmarshal(localProfileBytes, localProfile)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling failed: %v", err)
+	}
+	return localProfile, nil
+}
+
+func parseAndValidateLocalProfile(localProfileBytes []byte, getconfigCtx *getconfigContext) (*profile.LocalProfile, error) {
+	localProfile, err := parseLocalProfile(localProfileBytes)
+	if err != nil {
+		return nil, fmt.Errorf("parseAndValidateLocalProfile parseLocalProfile: %v", err)
+	}
+	if localProfile.GetServerToken() != getconfigCtx.profileServerToken {
+		// send something to ledmanager ??
+		return nil, fmt.Errorf("parseAndValidateLocalProfile: missamtch ServerToken for local profile server")
+	}
+	return localProfile, nil
+}
+
+// read saved local profile in case of particular reboot reason
+func readSavedLocalProfile(getconfigCtx *getconfigContext) (*profile.LocalProfile, error) {
+	localProfileMessage, err := readSavedProtoMessage(
+		getconfigCtx.zedagentCtx.globalConfig.GlobalValueInt(types.StaleConfigTime),
+		filepath.Join(checkpointDirname, savedLocalProfileFile), false)
+	if err != nil {
+		return nil, fmt.Errorf("lastlocalprofile: %v", err)
+	}
+	if localProfileMessage != nil {
+		log.Function("Using saved local profile")
+		return parseAndValidateLocalProfile(localProfileMessage, getconfigCtx)
+	}
+	return nil, errors.New("readSavedLocalProfile: empty message")
+}
+
+//prepareLocalProfileServerMap process configuration of network instances to find match with defined localServerURL
+//returns the srcIP and processed url for the zero or more network instances on which the localProfileServer might be hosted
+//based on a subnet inclusion match or hostname in dns records
+//in form bridge name -> slice of urlAndSrcIP
+func prepareLocalProfileServerMap(localServerURL string, getconfigCtx *getconfigContext) (map[string][]*urlAndSrcIP, error) {
+	u, err := url.Parse(localServerURL)
+	if err != nil {
+		return nil, fmt.Errorf("checkAndPrepareLocalIP: url.Parse: %s", err)
+	}
+	res := make(map[string][]*urlAndSrcIP)
+	appNetworkStatuses := getconfigCtx.subAppNetworkStatus.GetAll()
+	networkInstanceConfigs := getconfigCtx.pubNetworkInstanceConfig.GetAll()
+	localProfileServerHostname := u.Hostname()
+	localProfileServerIP := net.ParseIP(localProfileServerHostname)
+	for _, entry := range appNetworkStatuses {
+		status := entry.(types.AppNetworkStatus)
+		for _, config := range status.UnderlayNetworkList {
```

Comment:
Odd to name this config when it is from status. FWIW we use e.g., ulStatus elsewhere.